sdlib/d/gc/tcache.d

```diff
@@ -570,3 +571,31 @@ unittest extend {
 	// Extend by size zero still works, though:
 	assert(threadCache.extend(p1[0 .. 16384], 0));
 }
+
+unittest arraySpill {
+	void setUsed(void* ptr, size_t usedCapacity) {
+		assert(ptr !is null);
+		auto pd = threadCache.getPageDescriptor(ptr);
+		assert(pd.extent !is null);
+		assert(pd.extent.isLarge());
+		pd.extent.setUsedCapacity(usedCapacity);
+	}
+
+	// Verify that zero-length interstitial slice does not spill into next alloc:
+	void testAdjacentSpill(size_t arrSize, size_t a1Cap, size_t a2Cap) {
+		auto array1 = threadCache.alloc(arrSize, false);
+		auto array2 = threadCache.alloc(arrSize, false);
+		assert(array2 is array1 + arrSize);
```

Comment:
You have not done anything to ensure this is the case, and the state of the allocator depends on every other tests, so this is a recipe for spooky action at a distance.

About every single name in there is screwed up in some way.